tests/lib/rules/one-var.js

```diff
@@ -126,6 +182,83 @@ eslintTester.addRuleTest("lib/rules/one-var", {
                 message: "Combine this with the previous 'var' statement.",
                 type: "VariableDeclaration"
             }]
+        },
+        {
+            code: "function foo() { let a = 1; let b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "always"],
+            errors: [{
+                message: "Combine this with the previous 'let' statement.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "function foo() { const a = 1; const b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, "always"],
+            errors: [{
+                message: "Combine this with the previous 'const' statement.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "function foo() { let a = 1; let b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, {let: "always"}],
+            errors: [{
+                message: "Combine this with the previous 'let' statement.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "function foo() { const a = 1; const b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, {const: "always"}],
+            errors: [{
+                message: "Combine this with the previous 'const' statement.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "function foo() { let a = 1, b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, {let: "never"}],
+            errors: [{
+                message: "Split 'let' declaration into multiple statements.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "function foo() { const a = 1, b = 2; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, {const: "never"}],
+            errors: [{
+                message: "Split 'const' declaration into multiple statements.",
+                type: "VariableDeclaration"
+            }]
+        },
+        {
+            code: "let foo = true; switch(foo) { case true: let bar = 2; break; case false: let baz = 3; break; }",
+            ecmaFeatures: {
+                blockBindings: true
+            },
+            args: [2, {var: "always", let: "always", const: "never"}],
+            errors: [{
+                message: "Combine this with the previous 'let' statement.",
+                type: "VariableDeclaration"
```

Comment:
To foolproof this for any future changes, can you add a location to the expected error that will guarantee it's the `baz` declaration that's being reported here? ([Example](https://github.com/eslint/eslint/blob/ac77fd8b1b600779fb8ef25cead4df2807b6c024/tests/lib/rules/key-spacing.js#L169))